File: LeetCode/Explore/TopInterview/Blind75/String/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
          return False
        
        mp = Counter(s)
        
        for c in t:
          if c in mp and mp[c] > 0:
            mp[c] -= 1
          else:
            return False
          
        return True

File: LeetCode/Explore/TopInterview/Google/SortingSearching/242_valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
      if (s.size() != t.size()) {
        return false;
      }
      unordered_map<char, int> mp;
      for (auto &c : s) {
        mp[c]++;
      }
      for (auto &c : t) {
        if (mp.count(c) && mp[c] > 0) {
          mp[c]--;
        }
        else {
          return false;
        }
      }
      return true;
    }
};
